File: doc/src/scripts/tutorials/reference-frames/NavballDirections.cs
using System;
using System.Collections.Generic;
using System.Net;
using KRPC.Client;
using KRPC.Client.Services.SpaceCenter;

class NavballDirections
{
    public static void Main ()
    {
        using (var conn = new Connection ("Navball directions")) {
            var vessel = conn.SpaceCenter ().ActiveVessel;
            var ap = vessel.AutoPilot;
            ap.ReferenceFrame = vessel.SurfaceReferenceFrame;
            ap.Engage();

            // Point the vessel north on the navball, with a pitch of 0 degrees
            ap.TargetDirection = Tuple.Create (0.0, 1.0, 0.0);
            ap.Wait();

            // Point the vessel vertically upwards on the navball
            ap.TargetDirection = Tuple.Create (1.0, 0.0, 0.0);
            ap.Wait();

            // Point the vessel west (heading of 270 degrees), with a pitch of 0 degrees
            ap.TargetDirection = Tuple.Create (0.0, 0.0, -1.0);
            ap.Wait();

            ap.Disengage();
        }
    }
}

File: doc/src/scripts/tutorials/reference-frames/NavballDirections.lua
local krpc = require 'krpc'
local List = require 'pl.List'
local conn = krpc.connect('Navball directions')
local vessel = conn.space_center.active_vessel
local ap = vessel.auto_pilot
ap.reference_frame = vessel.surface_reference_frame
ap:engage()

-- Point the vessel north on the navball, with a pitch of 0 degrees
ap.target_direction = List{0, 1, 0}
ap:wait()

-- Point the vessel vertically upwards on the navball
ap.target_direction = List{1, 0, 0}
ap:wait()

-- Point the vessel west (heading of 270 degrees), with a pitch of 0 degrees
ap.target_direction = List{0, 0, -1}
ap:wait()

ap:disengage()
